### PR TITLE
Changed wazuh-agent service start from install test to registration test

### LINKED ISSUE #5741: `Job for wazuh-agent.service failed because` message was found during agent testing
| Target version | Related issue | Related PR/dev branch |
|---|---|---|
| 4.9.0  | https://github.com/wazuh/wazuh/issues/23748  | https://github.com/wazuh/wazuh-qa/pull/5689  |

The message:

```
It does not interfere with the tests themselves, but its nature should be investigated.
�[0;33m    Job for wazuh-agent.service failed because the control process exited with error code.�[0m
�[0;33m    See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.�[0m
```

It was found during agent testing in deployability tests (install process).

It does not interfere with the tests themselves, but its nature should be investigated.

### PR DESCRIPTION
**Pull request by hossam1522**
# Description

This pull request is based on https://github.com/wazuh/wazuh-qa/issues/5741. As said in https://github.com/wazuh/wazuh-qa/issues/5741#issuecomment-2364003587, the service cannot be started in the installation test because there is no manager IP assigned. Therefore, the solution is to start the service in the registration test, after adding the manager IP to the agent configuration.

After doing that and running the test, you can see in the logs that the message reported in the issue no longer appears:
- [agent.log](https://github.com/user-attachments/files/17079223/agent.log)

---

## Testing performed

| Validation | Jenkins | Local  | OS  | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|
|           | ⚫⚫ | :green_circle: :green_circle: |     Ubuntu 24.04    |    https://github.com/wazuh/wazuh-qa/commit/5584a69747aa5f1b01210855e43f3c2d37149b1d     | Nothing to highlight |

### CODE REVIEW

**Review by rauldpm — CHANGES_REQUESTED** [review]
The changes affect Windows and macOS systems, please run a complete agent tier

**Comment by hossam1522** [comment]
A complete agent tier test was executed correctly:

Log:
- [jobflow.log](https://github.com/user-attachments/files/17117842/jobflow.log)

Template:
- [template.txt](https://github.com/user-attachments/files/17117859/template.txt)

**Review by joaquinsgi — CHANGES_REQUESTED** [review]
Gj! But it is necessary to update the CHANGELOG.md

**Review by joaquinsgi — DISMISSED** [review]

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ! Please, solve the conflicts.

**Review by juliamagan — CHANGES_REQUESTED** [review]
Reviewing the test, I am a bit concerned about the implementation of this one:

1. It was failing to start but was not reported as failed.
2. The install starts the agent without registering it, which is why it fails. But also, it then checks for stopped, so it wouldn't make sense for it to start it, or for the check to be stopped.
3. The state check is too wide, it should check that the status is correct for all valid statuses, it would even be expected statuses instead of valid statuses.
4. The registration function is modified, which is not used by this test and this modification could affect where it is being used, since restart has been changed by start, and if the agent is already started, it is not going to be registered.


Please, let's take a quick look at this test, see what status is expected, see why the error is not being handled well and make the necessary modifications.

**Comment by hossam1522** [comment]
Everything related to starting the agent and checking its status has been removed from the installation function since, as no manager has been registered, it will fail to start causing the log reported in this issue. However, even if it fails in that part, the test does not report any failure and passes as valid, which should not happen and for which the following issue has been opened: 

- https://github.com/wazuh/wazuh-qa-automation/issues/1142

It has been considered most logical to start the agent after registering a manager. For this reason, the `restart` in the registration function has been modified by a `start`, because previously it was not possible to start the agent because there was no associated manager.

As this has been modified for Linux and MacOS only (The change that was made in the Windows test was tested on https://github.com/wazuh/wazuh-qa/pull/5762#issuecomment-2371816242), a build has been run to demonstrate that both the installation test and the registration test work perfectly and that the erroneous log reported in the issue no longer appears.

- Log: [jobflow.log](https://github.com/user-attachments/files/17262162/jobflow.log)

- Template: [template-5741.txt](https://github.com/user-attachments/files/17262170/template-5741.txt)

**Review by rauldpm — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,10 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Changed wazuh-agent service start from install test to registration test ([#5762](https://github.com/wazuh/wazuh-qa/pull/5762)) \- (Tests)
 - Fix E2E VD tests not working if the Python version is different to 3.9 ([#5787](https://github.com/wazuh/wazuh-qa/pull/5787)) \- (Framework)
-- Fix failed to disable wazuh-manager.service ([#5775])(https://github.com/wazuh/wazuh-qa/pull/5775) \- (Tests)
+- Fix failed to disable wazuh-manager.service ([#5775])(https://github.com/wazuh/wazuh-qa/pull/5775)) \- (Tests)
+
```
Comment (juliamagan) [inline CHANGELOG.md]:
```suggestion
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,8 +6,9 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Changed wazuh-agent service start from install test to registration test ([#5762](https://github.com/wazuh/wazuh-qa/pull/5762)) \- (Tests)
 - Fix E2E VD tests not working if the Python version is different to 3.9 ([#5787](https://github.com/wazuh/wazuh-qa/pull/5787)) \- (Framework)
-- Fix failed to disable wazuh-manager.service ([#5775])(https://github.com/wazuh/wazuh-qa/pull/5775) \- (Tests)
+- Fix failed to disable wazuh-manager.service ([#5775])(https://github.com/wazuh/wazuh-qa/pull/5775)) \- (Tests)
```
Comment (juliamagan) [inline CHANGELOG.md]:
```suggestion
- Fix failed to disable wazuh-manager.service ([#5775](https://github.com/wazuh/wazuh-qa/pull/5775)) \- (Tests)
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

**Review by juliamagan — APPROVED** [review]

**Pull request merged**